client/checkout/utils/upe.js

```diff
@@ -300,3 +300,71 @@ export const getStripeElementOptions = (
 
 	return options;
 };
+
+/**
+ * Converts form fields object into Stripe `shipping` object.
+ *
+ * @param {Object} fields Object mapping checkout shipping fields to values.
+ * @return {Object} Stripe formatted `shipping` object.
+ */
+export const getShippingDetails = ( fields ) => {
+	// Shipping address is needed by Afterpay. If available, use shipping address, else fallback to billing address.
+	if (
+		document.getElementById( 'ship-to-different-address-checkbox' ) &&
```

Comment:
>Also, did we test blocks integration here?

I have a separate PR for [block checkout](https://github.com/Automattic/woocommerce-payments/pull/6838). Did you mean that? or perhaps some other arrangement?

Reply:
Yes, that one.

Reply:
>How reliable is picking the element ID in the context of WP customizations, things like Elementor?

This is a good point. I will check this deeper.  I saw a similar check in the existing code though [for StripeLink](https://github.com/Automattic/woocommerce-payments/blob/a3039cdca03d6fcb41f2cd720757c4601e9ea46a/client/checkout/classic/upe.js#L268). 

Let me do a check on Elementor, or find a better way to trace this 👍

Reply:
Worth re-evaluating, thanks.

Reply:
@kalessil  - I did a deeper check with external themes like Flatsome ( proprietary ), and Astra ( Elementor and Beaver Builder versions ). I found that it is better to use the checkbox form submit field instead of the CSS ID. It seems to work correctly across all the themes I mentioned above, and Storefront too. 

I have made the change here, and I hope this is a good solution
f0592fc

Reply:
That looks much better, thanks!